transaction_test.go

```diff
@@ -35,3 +38,65 @@ func TestTransactionOptions(t *testing.T) {
 		t.Fatalf("should have returned Snowflake Error: %v", errMsgNoDefaultTransactionIsolationLevel)
 	}
 }
+
+// SNOW-823072: Test that transaction uses the context object supplied by BeginTx(), not from the parent connection
+func TestTransactionContext(t *testing.T) {
+	var tx *sql.Tx
+	var err error
+
+	db := openDB(t)
+	defer db.Close()
+
+	ctx1, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+	defer cancel()
+
+	ctx2 := context.Background()
+
+	pingWithRetry := withRetry(PingFunc, 5, 3*time.Second)
+
+	err = pingWithRetry(ctx1, db)
```

Comment:
This should use `ctx2`, not `ctx1` since the `withRetry` helper function sets the timeout to 3 seconds automatically.  You can actually get rid of `ctx1`, but given the retry logic, you may end up trying to create a context from a base context that has already timed out.

Reply:
I will update this in a separate PR, I missed the edit on the repro. Thanks @rohantandon